docs/site/Deploying-with-pm2-and-nginx.md

```diff
@@ -0,0 +1,93 @@
+---
+lang: en
+title: 'Deploying with pm2 and nginx'
+keywords: LoopBack, pm2, nginx
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/deploying-with-pm2-and-nginx.html
+---
+
+This is a basic guide for deploying a LoopBack 4 (LB4) app using
+[pm2](https://www.npmjs.com/package/pm2) behind nginx reverse proxy.
+
+PM2 is a Production Runtime and Process Manager for Node.js applications with a
+built-in Load Balancer. It allows you to keep applications alive forever, to
+reload them without downtime and facilitate common Devops tasks.
+
+NGINX is open source software for web serving, reverse proxying, caching, load
+balancing, media streaming, and more. It started out as a web server designed
+for maximum performance and stability. In addition to its HTTP server
+capabilities, NGINX can also function as a proxy server for email (IMAP, POP3,
+and SMTP) and a reverse proxy and load balancer for HTTP, TCP, and UDP servers.
+
+**NOTE**: Production deployment is a detailed topic and explained with clarity
+in their respective documentations. Please refer [pm2 docs](https://pm2.io/doc/)
+and
+[nginx docs](https://docs.nginx.com/nginx/admin-guide/installing-nginx/installing-nginx-open-source/)
+for detailed setup instructions. This guide assumes you have nginx setup
+already.
+
+## First steps
+
+Before we start with deployment, let's get our app ready. If you have loopback
+cli installed, run the following command to create a new app:
+
+```sh
+$ lb4 app
+```
+
+Refer to
+[getting started](https://loopback.io/doc/en/lb4/Getting-started.html#create-a-new-project)
+section of documentation for detailed instructions.
+
+You also need to have pm2 installed globally. If not you don't already have pm2
```

Comment:
It a global install something that `pm2` recommends? In my experience, it's usually better to use a local install and invoke `pm2` via npm script.

```json
{
  "scripts": {
    "start": " pm2 start index.js"
  },
  "dependencies": {
    "pm2": "^1.2.3"
  }
}
```

That way each app retains control over which version of `pm2` it is working with, and developers working on multiple apps don't run into a possible conflict in the required version of the global `pm2` install.

Reply:
Sounds good, thanks 👍